src/software/geom/algorithms/distance.cpp

```diff
@@ -66,6 +66,16 @@ double distance(const Circle &first, const Point &second)
     return distance(second, first);
 }
 
+double distance(const Point &first, const Stadium &second)
+{
+    return fmax(distance(first, second.segment()) - second.radius(), 0);
```

Comment:
nit: we generally tend to use C++ STL implementation of functions, rather than C ones
```suggestion
    return std::max(distance(first, second.segment()) - second.radius(), 0);
```